src/Nethermind/Nethermind.Consensus.AuRa/AuRaMultiSyncModeSelector.cs

```diff
@@ -0,0 +1,52 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using Nethermind.Blockchain.Synchronization;
+using Nethermind.Logging;
+using Nethermind.Synchronization.ParallelSync;
+using Nethermind.Synchronization.Peers;
+
+namespace Nethermind.Consensus.AuRa
+{
+    public class AuRaMultiSyncModeSelector : MultiSyncModeSelector
+    {
+        public AuRaMultiSyncModeSelector(ISyncProgressResolver syncProgressResolver, ISyncPeerPool syncPeerPool, ISyncConfig syncConfig, ILogManager logManager) 
+            : base(syncProgressResolver, syncPeerPool, syncConfig, logManager)
+        {
+        }
+        
+        protected override bool AnyDesiredPeerKnown(Snapshot best)
+        {
+            if (base.AnyDesiredPeerKnown(best))
+            {
+                // we really, really don't trust parity when its saying it has same block level with higher difficulty in AuRa, its lying most of the times in AuRa
```

Comment:
in CLique too

Reply:
Adding comments, this may be AuRa specific. Not sure how difficulty is calculated for clique